### PR TITLE
data-parallel patched ALP standalone kernel

### PR DESCRIPTION
**Pull request by a10y**
## Summary

Follow up to #7440 

This changes ALP execution on CUDA. Previously, we'd execute two kernel passes: one to perform ALP decoding to global memory, and a second to apply patches.

This PR works similarly to prior work to push patching into the decoding kernel for unpacking. We assign a FastLanes 1024-element block to each warp (32 threads), and then perform decoding and patching in a single kernel pass.

## Testing

Unit tests were added to check for simple and edge cases (multi-chunk, mix of chunks with/without patches)

### CODE REVIEW

**Comment by a10y** [comment]
Benchmark improvements VS develop on GH 200

```
~/vortex$ nvidia-smi
Mon Apr 20 22:09:35 2026
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.148.08             Driver Version: 570.148.08     CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GH200 480GB             On  |   00000000:DD:00.0 Off |                  Off |
| N/A   35C    P0             74W /  700W |       3MiB /  97871MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```

Screenshot 2026-04-20 at 18 08 42

File: vortex-cuda/src/kernel/encodings/alp.rs [hunk 1]
```diff
@@ -257,4 +276,179 @@ mod tests {
...
+            prim.as_view(),
+            None,
+            &mut LEGACY_SESSION.create_execution_ctx(),
+        )?;
+        assert!(
+            alp_array.patches().is_some(),
+            "expected patches from ALP exceptions"
+        );
+
+        let cpu_result = crate::canonicalize_cpu(alp_array.clone())?.into_array();
```
Comment (robert3005) [inline vortex-cuda/src/kernel/encodings/alp.rs:351]: why no import?

Comment (a10y) [reply vortex-cuda/src/kernel/encodings/alp.rs]: fixed

File: vortex-cuda/kernels/src/alp.cu [hunk 2]
```diff
@@ -1,36 +1,92 @@
...
+//
+// Each block processes one 1024-element chunk cooperatively and applies patches
+// into shared memory before writing to global memory, mirroring the strategy
+// used by bit_unpack. f = F10[exponents.f], e = IF10[exponents.e].
+//
+// The cast from EncT to FloatT must preserve ALP's lossless contract: f32 is
+// only encoded as i32, and f64 is only encoded as i64. The i64 → double cast
+// is lossless for all values ALP can produce.
+template <typename EncT, typename FloatT>
+__device__ void _alp_device(const EncT *__restrict in,
```
Comment (0ax1) [inline vortex-cuda/kernels/src/alp.cu]: why the _ prefix?

Comment (a10y) [reply vortex-cuda/kernels/src/alp.cu]: that was claude :P

**Review by 0ax1 — APPROVED** [review]
